WordPress/Classes/Services/MediaService.m

```diff
@@ -56,6 +60,141 @@ - (id)initWithManagedObjectContext:(NSManagedObjectContext *)context
     return self;
 }
 
+- (void)syncMediaLibraryForBlog:(Blog *)blog
+                        success:(void (^)())success
+                        failure:(void (^)(NSError *error))failure
+{
+    id<MediaServiceRemote> remote = [self remoteForBlog:blog];
+    [remote getMediaLibraryForBlog:blog
+                           success:^(NSArray *media) {
+                               [self.managedObjectContext performBlock:^{
+                                   [self mergeMedia:media forBlog:blog completionHandler:success];
+                               }];
+                           }
+                           failure:^(NSError *error) {
+                               if (failure) {
+                                   [self.managedObjectContext performBlock:^{
+                                       failure(error);
+                                   }];
+                               }
+                           }];
+}
+
+- (void)mergeMedia:(NSArray *)media forBlog:(Blog *)blog completionHandler:(void (^)(void))completion
+{
+    NSMutableArray *mediaToKeep = [NSMutableArray array];
+    for (RemoteMedia *remote in media) {
+        Media *local = [self findMediaWithID:remote.mediaID inBlog:blog];
+        if (!local) {
+            local = [self newMediaForBlog:blog];
+            local.remoteStatus = MediaRemoteStatusSync;
+        }
+        [self updateMedia:local withRemoteMedia:remote];
+        [mediaToKeep addObject:local];
+    }
+    
+    NSSet *existingMedia = blog.media;
+    NSMutableOrderedSet *mediaToThumbnail = [NSMutableOrderedSet orderedSet];
+    if (existingMedia.count > 0) {
+        for (Media *existing in existingMedia) {
+            if (![mediaToKeep containsObject:existing] && existing.remoteURL != nil) {
+                [self.managedObjectContext deleteObject:existing];
+            } else if (existing.remoteURL.length && !existing.thumbnail && existing.mediaType == MediaTypeImage) {
+                [mediaToThumbnail addObject:existing];
+            }
+        }
+    }
+    
+    [[ContextManager sharedInstance] saveDerivedContext:self.managedObjectContext];
+    
+    [mediaToThumbnail sortUsingComparator:^NSComparisonResult(Media *left, Media *right) {
+        return [right.creationDate compare:left.creationDate];
+    }];
+    [self getThumbnailsForMedia:mediaToThumbnail];
```

Comment:
I know this is pushed to a background thread with low priority, but should we started pulling thumbnails for the blog on the background when we sync a blog? this could be a big waste of the user data plan for something that we may not use  immediately.